src/microbe_stage/systems/SiderophoreSystem.cs

```diff
@@ -0,0 +1,137 @@
+    ///   to make doubly sure this callback won't be garbage collected while the native side still has a reference to
+    ///   it.
+    /// </summary>
+    private readonly IWorldSimulation worldSimulation;
+
+    public SiderophoreSystem(World world, IParallelRunner runner, IWorldSimulation worldSimulation) :
+        base(world, runner)
+    {
+        this.worldSimulation = worldSimulation;
+    }
+
+    protected override void Update(float delta, in Entity entity)
+    {
+        if (!entity.Has<SiderophoreProjectile>())
+            return;
+
+        ref var projectile = ref entity.Get<SiderophoreProjectile>();
+
+        if (projectile.IsUsed)
+            return;
+
+        ref var collisions = ref entity.Get<CollisionManagement>();
+
+        if (!projectile.ProjectileInitialized)
+        {
+            projectile.ProjectileInitialized = true;
+
+            collisions.StartCollisionRecording(Constants.MAX_SIMULTANEOUS_COLLISIONS_TINY);
+        }
+
+        // Check for active collisions that count as a hit and use up this projectile
+        var count = collisions.GetActiveCollisions(out var activeCollisions);
+        for (int i = 0; i < count; ++i)
+        {
+            ref var collision = ref activeCollisions![i];
+
+            if (!HandleSiderophoreCollision(ref collision, in worldSimulation, entity
+                    .Get<SiderophoreProjectile>().Sender))
+            {
+                continue;
+            }
+
+            // Applied a damaging hit, destroy this toxin
+            // TODO: We should probably get some *POP* effect here.
+
+            // Expire right now
+            ref var timedLife = ref entity.Get<TimedLife>();
+            timedLife.TimeToLiveRemaining = -1;
+
+            ref var physics = ref entity.Get<Physics>();
+
+            // TODO: should this instead of disabling the further collisions be removed from the world immediately
+            // to cause less of a physics impact?
+            // physics.BodyDisabled = true;
+            physics.DisableCollisionState = Physics.CollisionState.DisableCollisions;
+
+            // And make sure the flag we check for is set immediately to not process this projectile again
+            // (this is just extra safety against the time over callback configuration not working correctly)
+            projectile.IsUsed = true;
+
+            // Only deal damage at most to a single thing
+            break;
+        }
+    }
+
+    private static bool HandleSiderophoreCollision(ref PhysicsCollision collision,
+        in IWorldSimulation worldSimulation, Entity sender)
+    {
+        var target = collision.SecondEntity;
+
+        // Skip if hit something that's not a chunk
+        if (!target.Has<ChunkConfiguration>())
+            return false;
+
+        ref var configuration = ref target.Get<ChunkConfiguration>();
+
+        var iron = SimulationParameters.Instance.GetCompound("iron");
```

Comment:
Iron should be stored as a field in this class (also means that this method cannot be static anymore).